app/views/posts/index.html.erb

```diff
@@ -0,0 +1,29 @@
+<p id="notice"><%= notice %></p>
+
+<h1>Posts</h1>
+
+
+<table>
+  
+
+  <tbody>
+    <% @posts.each do |post| %>
+      <div>
+      <p>Post title:<%= post.title%></p>
+      <p>Post body:<%= post.body%></p>
+      <p>Id:<%= post.id%></p>
+      <%= show_name(post) %>
+      </div>
+    <% end %>
+  </tbody>
+</table>
+
+<br>
+<% if user_signed_in?%>
+<p><%= link_to current_user.name, edit_user_registration_path %> </p>
+<p><%= link_to 'New Post', new_post_path %></p>
+<p><%= link_to 'Sign out', destroy_user_session_path, method: :delete %></p>
+<%else%>
+<p><%= link_to 'Sign in', new_user_session_path %> </p>
+<p><%= link_to 'Sign Up', new_user_registration_path%> </p>
+<%end%>
```

Comment:
Kindly move this logic to helpers. Views should be strictly used for rendering information from the controller and not generating one